### PR TITLE
Error codes in a separate file. Closes #13

### LINKED ISSUE #13: Export errors codes to allow translation
#### Support plan

* *which support plan is this issue covered by?* (e.g. Community, Core, Plus, or Enterprise): Free support 
* *is this issue currently blocking your project?* (yes/no): no
* *is this issue affecting a production system?* (yes/no): no

#### Context

* *node version*: v12.11.1
* *module version*: latest
* *environment* (e.g. node, browser, native): browser
* *used with* (e.g. hapi application, another framework, standalone, ...): standalone
* *any other relevant information*: Typescript 3.6.2

#### What problem are you trying to solve?

Hi, thanks for the template and the code of conduct - very inspirational words. Hope everything will be ok with this issue.

I'd like to use this `address` module: it is very exhaustive and I love the detailed feedback it gives back in case of error.

But I also need to translate error codes so I would like to have a list of all errors used and / or error codes. In my case have a translation/error-codes is mandatory and can compromise to use or not this library.

#### Do you have a new or modified API suggestion to solve the problem?

I'm here to discuss what is the best way to do it (if this can be a good idea for the maintainer) and hopefully find the time to have a mini pull-request.

One idea can be:

```js

const errorsByCode = {
    BE_STRING: 'Invalid input: email must be a string',
	BE_NON_EMPTY_STRING: 'Address must be a non-empty string',
    // ....
};

module.exports = {
	errorsByCode,
    domain: {
        // ....
    },
    email: {
        // ....       
    }
};

// use the error-codes in the rest of the source code
```

What do you think?

### PR DESCRIPTION
**Pull request by nkint**
Following comments in #13 .
I'm not totally sure if types test are correct.
Let me know if requires further changes, thank you.

### CODE REVIEW

File: lib/domain.js [hunk 1]
```diff
@@ -16,28 +18,28 @@ const internals = {
 exports.analyze = function (domain, options = {}) {
 
     if (typeof domain !== 'string') {
-        throw new Error('Invalid input: domain must be a string');
+        throw new Error(errorsByCode.DOMAIN_BE_STRING);
```
Comment (hueniverse) [inline lib/domain.js:21]: This is developer error, not a validation error. Leave that as a string.

File: lib/domain.js [hunk 2]
```diff
@@ -16,28 +18,28 @@ const internals = {
...
     }
 
     if (!domain) {
-        return { error: 'Domain must be a non-empty string' };
+        return { error: errorsByCode.DOMAIN_NON_EMPTY_STRING };
```
Comment (hueniverse) [inline lib/domain.js]: If we are going to do this, we should add actual codes. E.g. `{ error: errorsByCode.DOMAIN_NON_EMPTY_STRING, code: 'DOMAIN_NON_EMPTY_STRING' }`

**Review by hueniverse — CHANGES_REQUESTED** [review]

File: lib/domain.js (hunk 1 above)
Comment (nkint) [reply lib/domain.js:21]: Ok. I think the same should be applied to the TLDS option validation.

File: lib/domain.js (hunk 2 above)
Comment (nkint) [reply lib/domain.js]: Ok

**Pull request merged**